### PR TITLE
Use a FileTracker for transitive html analysis, but can't yet do dart…

### PR DESCRIPTION
**Pull request by MichaelRFairhurst**
… signatures, in practice

### CODE REVIEW

File: analyzer_plugin/lib/src/angular_driver.dart [hunk 1]
```diff
@@ -72,7 +81,23 @@ class AngularDriver
 
   void fileChanged(String path) {
     if (_ownsFile(path)) {
-      _changedFiles.add(path);
+      if (path.endsWith('.html')) {
+        for (final dartContext
+            in _fileTracker.getDartPathsReferencingHtml(path)) {
+          _htmlViewsToAnalyze.add(new Tuple2(path, dartContext));
+        }
+        for (final path in _fileTracker.getHtmlPathsReferencingHtml(path)) {
```
Comment (scheglov) [inline analyzer_plugin/lib/src/angular_driver.dart]: Ouch.

File: analyzer_plugin/lib/src/angular_driver.dart [hunk 2]
```diff
@@ -377,7 +373,9 @@ class AngularDriver
   Future pushDartOccurrences(String path) async {}
 
   Future pushDartErrors(String path) async {
-    final errors = (await resolveDart(path)).errors;
+    final result = (await resolveDart(path));
```
Comment (scheglov) [inline analyzer_plugin/lib/src/angular_driver.dart]: You can remove parenthesis here.

File: analyzer_plugin/lib/src/file_tracker.dart [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  _RelationshipTracker _dartToHtml = new _RelationshipTracker();
+
+  Set<String> _dartFilesWithDartTemplates = new HashSet<String>();
+
+  void setDartHtmlTemplates(String dartPath, List<String> htmlPaths) {
+    return _dartToHtml.setFileReferencesFiles(dartPath, htmlPaths);
+  }
+
+  void setDartHasTemplate(String dartPath, bool hasTemplate) {
+    if (!hasTemplate) {
```
Comment (scheglov) [inline analyzer_plugin/lib/src/file_tracker.dart]: I'd invert this if statement.

File: analyzer_plugin/lib/src/angular_driver.dart (hunk 1 above)
Comment (MichaelRFairhurst) [inline analyzer_plugin/lib/src/angular_driver.dart]: Haha, you mean ouch that we have to track this? I so agree :)

**Pull request merged**